src/reader.rs

```diff
@@ -28,9 +28,15 @@ pub struct Reader<'t, D: Distance> {
 impl<'t, D: Distance> Reader<'t, D> {
     pub fn open<U>(
         rtxn: &'t RoTxn,
-        index: u16,
+        index: u8,
```

Comment:
Can you add an infromation on what's the limit of this `u8`, please? Just a single doc line, we will refine that later.